RealmSwift/SwiftUI.swift

```diff
@@ -1691,7 +1689,11 @@ private class ObservableAsyncOpenStorage: ObservableObject {
                  user's sync configuration for the given partition value will be set as the `syncConfiguration`,
                  if empty the user configuration will be used.
      - parameter timeout: The maximum number of milliseconds to allow for a connection to
-     become fully established., if empty or `nil` no connection timeout is set.
+                 become fully established., if empty or `nil` no connection timeout is set.
+
+     - note: It is recommend to use the configuration obtained from the user, using `user.configuration(partitionValue:)`
+     or `user.flexibleSyncConfiguration()` when adding a configuration to this property wrapper, and modify any option from it
+     if needed.
```

Comment:
Related to some issue https://github.com/realm/realm-swift/issues/7931), since v10.12.0, but now that I think about it, we should disable any use of a configuration which doesn't come from a user. And avoid this issues when users inject a local configuration so they can set options like encryptedKey or fileUrl.